crates/bevy_pbr/src/render/pbr_fragment.wgsl

```diff
@@ -41,11 +41,13 @@ fn pbr_input_from_vertex_output(
     pbr_input.material.base_color = in.color;
 #endif
 
+#ifdef VERTEX_NORMALS
```

Comment:
this also looks wrong to me. the function is meant to be used either
- in the main pass, and vertex normals are always available in the forward_io vertex output
- in a deferred prepass, where `prepass_io::VertexOutput::world_normal` is gated on `NORMAL_PREPASS_OR_DEFERRED_PREPASS`, (so is always available in a deferred prepass)

again it's not harmful (vertex normals are required for lit materials to work, and the pbr_input normal is not used for unlit materials) but not helpful either. if you're hitting an error here i guess you're using the function in a prepass, which isn't intended except in deferred.

not blaming you for misunderstanding - there's definitely a problem with lack of documentation and help for this area - but if you want to share what you're trying to do we can probably work out the right approach (or i'm wrong, and then i will learn something).

Reply:
i'm speculating that you're not using deferred and you've used your main pass fragment shader for the prepass_fragment_shader as well?

Reply:
Yes, I was using the same for both:

```rust
impl MaterialExtension for FloorNoiseMaterial {
    fn fragment_shader() -> ShaderRef {
        "shaders/floor_noise.wgsl".into()
    }

    fn prepass_fragment_shader() -> ShaderRef {
        "shaders/floor_noise.wgsl".into()
    }
}
```
Mainly because I couldn't figure out how to define a separate fragment entry point without creating a separate asset for it. If I don't define the prepass_fragment_shader, then I get no shadows.

Reply:
If you only care about shadows and you don’t use any discards, you don’t need a prepass fragment shader at all. The vertex shader alone is enough to get depth info, and that can be the same as your main pass vertex shader. 

If you do use discards you’ll need to copy enough of the main pass frag shader code to determine if you need to discard, but still don’t need to return anything. 

If you need other prepass outputs (normals, motion vecs) then you need to explicitly return them, but iirc normals should not be normal mapped so you just return the vertex normal (or a VertexOuput with default values if you don’t even have vertex normals). Both normal and motion vec output you can copy/paste from pbr_prepass.wgsl 99% of the time.